roachprod: add tenant param to cluster.Conn* methods

msbutler opened the pull request:
This patch allows the roachtest writer to easily open a sql connection with an
in-process tenant.

Release note: None

Epic: None.

msbutler committed:
roachprod: add tenant param to InternalPGURL and ExternalPGURL methods

This patch will make it easier for roachtest writers to establish a sql
connection with an in-process tenant.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95586)
<!-- Reviewable:end -->

msbutler commented:
the patch works in the example in my last commit.

msbutler commented:
unrelated CI test flake, filed issue

stevendanna commented on pkg/cmd/roachtest/cluster.go:
```diff
@@ -2299,8 +2300,10 @@ func (c *clusterImpl) ExternalIP(
-func (c *clusterImpl) Conn(ctx context.Context, l *logger.Logger, node int) *gosql.DB {
-	urls, err := c.ExternalPGUrl(ctx, l, c.Node(node))
+func (c *clusterImpl) Conn(
+	ctx context.Context, l *logger.Logger, node int, tenant string,
```
I wonder if this might be a place where we want to use the "functional options" pattern so that we could do something like:

```golang
func (c *clusterImpl) Conn(ctx context.Context, l *logger.Logger, node int, opts ...ConnectionOption)
```

such that we don't have to update every single caller.

msbutler replied:
aha, this is a good idea. I'll plan to refactor with `...ConnectionOptions` pattern as it will take 5 minutes to refactor with Goland's fancy refactor tool. I'll wait until @smg260 and/or @renatolabs comment on this API change though.

renatolabs replied:
Definitely a 👍 from me.

msbutler replied:
function options are _slick_

stevendanna reviewed (commented):
Thanks for handling this.  I have one idea that might avoid updating every caller. But, since you've already done the work of updating all the callers, maybe it isn't worth it.

smg260 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 7 files at r1, 88 of 88 files at r4, 88 of 88 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95586#-:-NMZmUAoCqBDZ3qd8kQz:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @renatolabs)



<!-- Sent from Reviewable.io -->

msbutler committed:
roachprod: add functional connectionOptions to cluster.Conn methods

This patch makes it easier to add various options to the cluster.Conn methods,
such as easily allowing a roachtest writer to open a sql connection to an in
process tenant or as a specific user.

Epic: None

Release note: None

msbutler committed:
c2c: create role within tenant in roachtest

Release note: None

Epic: None

msbutler commented:
TFTR!

bors r=smg260

msbutler commented:
bors retry
